src/DotNetWorker.Grpc/Definition/GrpcFunctionDefinition.cs

```diff
@@ -79,8 +84,83 @@ public GrpcFunctionDefinition(FunctionLoadRequest loadRequest, IMethodInfoLocato
+                var triggerAttribute = parameterInfo?.GetCustomAttribute<TriggerBindingAttribute>();
+
+                return GetBindingAttributePropertiesDictionary(inputAttribute) ??
+                        GetBindingAttributePropertiesDictionary(triggerAttribute) ??
+                        ImmutableDictionary<string, object>.Empty;
+            }
+        }
+
+        private ImmutableDictionary<string, object>? GetBindingAttributePropertiesDictionary(BindingAttribute? bindingAttribute)
+        {
+            if (bindingAttribute is null)
+            {
+                return null;
+            }
+
+            var output = new Dictionary<string, object>();
+            bool isInputConverterAttributeAdvertised = false;
+
+            // ConverterTypesDictionary will be "object" part of the return value - ImmutableDictionary<string, object>
+            // The dictionary has key of type IInputConverter and value as List of Types supported by the converter.
+            var converterTypesDictionary = new Dictionary<Type, List<Type>>();
+
+
+            Type type = bindingAttribute.GetType();
+            var attributes = type.GetCustomAttributes<InputConverterAttribute>();
+
+            if (attributes.Any())
+            {
+                isInputConverterAttributeAdvertised = true;
+
+                foreach (var attribute in attributes)
+                {
+                    Type converter = attribute.ConverterType;
+                    List<Type> supportedTypes = GetTypesSupportedByConverter(converter);
+                    converterTypesDictionary.Add(converter, supportedTypes);
+                }
+            }
+
+            output.Add(PropertyBagKeys.BindingAttributeSupportedConverters, converterTypesDictionary);
+
+            if (isInputConverterAttributeAdvertised)
+            {
+                output[PropertyBagKeys.AllowConverterFallback] = type.GetCustomAttribute<AllowConverterFallbackAttribute>()?.AllowConverterFallback ?? true;
+            }
+
+            return output.ToImmutableDictionary();
+        }
+
+        private List<Type> GetTypesSupportedByConverter(Type converter)
+        {
+            var types = new List<Type>();
+
+            foreach (CustomAttributeData converterAttribute in converter.CustomAttributes)
+            {
+                if (converterAttribute.AttributeType == typeof(SupportedConverterTypeAttribute))
+                {
+                    foreach (CustomAttributeTypedArgument supportedType in converterAttribute.ConstructorArguments)
+                    {
+                        if (supportedType.ArgumentType is not null
+                            && supportedType.Value is not null
+                            && supportedType.ArgumentType == typeof(Type))
+                        {
+                            Type? supportedTypeValue = supportedType.Value as Type;
+
+                            if (supportedTypeValue is not null)
```

Comment:
```suggestion
                            if (supportedType.Value is Type type)
```
Can remove the `as` check above.